### DISCUSSION TITLE
wifi not showing up in Klipperscreen 🤔

### DISCUSSION THREAD

**Discussion by Buko84** [xblax/flashforge_ad5m_klipper_mod#69] [Q&A] [answered]
Hi, I just installed this mod, it's super great! I was able to finally level up my bed but I am getting problems with wifi, no networks are showing up in More / Network. I tried with going out from that menu and then back but nothing helped 🤔 I have 5M Pro.
![IMG_1086](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/157248367/0cf85bdb-0632-4b07-9a55-d2b80859d778)

**Comment by consp** [comment]
See #38 it's a known issue. I've had it once on the non-pro but maybe it's a pro issue.

If you know how to log with ssh in on the device can you perform the following and report the result:

```
iwctl device list
```

it should provide a wlan[number] device like wlan0

then with the device found in the list and replace wlan0 with it if it is different:
```
iwctl station wlan0 get-networks
```
Note: if you are not comfortable posting the network names, please just provide an answer if it reports anything at all.

If possible could you post the logs from `/var/log/KlipperScreen.log` after opening the wifi screen.

There is a known issue (not yet reported with an actual issue) with non ascii characters in the SSID names. Do you have any in range with non standard characters? Or spaces? Reason I found out it's an issue since someone in my neighborhood added utf8 emoticons to the network name yesterday.

**Reply by Buko84** [reply]
i will do it tomorrow, currently out of home :/

**Comment by Buko84** [comment]
![image](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/157248367/acc29090-6aca-42eb-b9da-00479c45b549)
it is showing networks but not in klipperscreen :/ 

![image](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/157248367/b1c37d2c-1bde-43c3-af23-c7c1820aa356)

**Reply by consp** [reply]
> it is showing networks but not in klipperscreen :/

I'm not surprized, it's a bit of a hack since it's not supported as-is. It's a work in progress.

Thank you for the data, should be helpful, does any of the wifi channels have spaces or non-ascii characters in them (like katakana)?

**Reply by Buko84** [reply]
nope, all names are just normal letters and numbers. some of them have _ between

**Reply by consp** [reply]
Looks like I made an error in the regex somewhere. Back to the drawingboard on that one.

Do you know how to upload a file to the machine? (e.g. with scp or winscp)

**Reply by Buko84** [reply]
yes I can do that :)

**Reply by consp** [reply]
I'm going to test some code (when my print finishes) and if I'm happy with it I'll send you a replacement wifi_iwd.py file. Maybe that will help (and otherwise print more data).

**Reply by consp** [reply]
If you could get this file: https://raw.githubusercontent.com/consp/KlipperScreen/wifi-fix-test/ks_includes/wifi_iwd.py

curl is installed on the printer so this should work: 
```
curl "https://raw.githubusercontent.com/consp/KlipperScreen/wifi-fix-test/ks_includes/wifi_iwd.py" -o ~/printer_software/KlipperScreen/ks_includes/wifi_iwd.py
```
and then `/etc/init.d/S80klipperscreen restart`.

Best case: it works now. Worst case more logging.

**Reply by Buko84** [reply, marked as answer]
works fine now! 💯 🥇

**Reply by consp** [reply]
Good to heat, then I'll set it up for a merge and mark it as solved.
